app/models/notification.rb

```diff
@@ -41,9 +41,11 @@ class Notification < ApplicationRecord
   validates :account_id, uniqueness: { scope: [:activity_type, :activity_id] }
   validates :activity_type, inclusion: { in: TYPE_CLASS_MAP.values }
 
-  scope :browserable, ->(exclude_types = []) {
+  scope :browserable, ->(exclude_types = [], account_id = nil) {
     types = TYPE_CLASS_MAP.values - activity_types_from_types(exclude_types + [:follow_request])
-    where(activity_type: types)
+    return where(activity_type: types) if account_id.nil?
```

Comment:
This is a lambda, not a method, so the `return` keyword acts differently here. I suggest just an `if/else` block. Or you can change the scope to a class-level method:

```ruby
class << self
  def browserable(exclude_types = [], account_id = nil)
    types = TYPE_CLASS_MAP.values - activity_types_from_types(exclude_types + [:follow_request])
    return where(activity_type: types) if account_id.nil?
    where(activity_type: types, from_account_id: account_id)
  end
end
```

Reply:
done, though I thought `return` acted differently in procs not lambdas